django/core/handlers/base.py

```diff
@@ -228,44 +175,88 @@ def get_response(self, request):
-                        "%s.process_response didn't return an "
-                        "HttpResponse object. It returned None instead."
-                        % (middleware_method.__self__.__class__.__name__))
+            # This is a noop with new style middlewares!
+            response = self._apply_old_response_middleware(request, response)
             response = self.apply_response_fixes(request, response)
         except Exception:  # Any exception should be gathered and handled
             signals.got_request_exception.send(sender=self.__class__, request=request)
-            response = self.handle_uncaught_exception(request, resolver, sys.exc_info())
+            response = self.handle_uncaught_exception(request, sys.exc_info())
 
         response._closable_objects.append(request)
 
         # If the exception handler returns a TemplateResponse that has not
         # been rendered, force it to be rendered.
-        if not response_is_rendered and callable(getattr(response, 'render', None)):
+        if not getattr(response, 'is_rendered', True) and callable(getattr(response, 'render', None)):
+
             response = response.render()
 
         return response
 
-    def process_exception_by_middleware(self, exception, request):
-        """
-        Pass the exception to the exception middleware. If no middleware
-        return a response for this exception, raise it.
-        """
-        for middleware_method in self._exception_middleware:
-            response = middleware_method(request, exception)
+    def _get_response(self, request):
+        "Returns an HttpResponse object for the given HttpRequest"
+        response = None
+
+        if hasattr(request, 'urlconf'):
+            urlconf = request.urlconf
+            set_urlconf(urlconf)
+            resolver = get_resolver(urlconf)
+        else:
+            resolver = get_resolver()
+
+        resolver_match = resolver.resolve(request.path_info)
+        callback, callback_args, callback_kwargs = resolver_match
+        request.resolver_match = resolver_match
+
+        # Apply view middleware
+        for middleware_method in self._view_middleware:
+            response = middleware_method(request, callback, callback_args, callback_kwargs)
             if response:
                 return response
-        raise
 
-    def handle_uncaught_exception(self, request, resolver, exc_info):
+        wrapped_callback = self.make_view_atomic(callback)
+
+        # TODO: It would be nice if we could move the try/except in _get_response_old,
+        # but that would break semantics.
+        try:
+            response = wrapped_callback(request, *callback_args, **callback_kwargs)
+        except Exception as e:
+            response = self.process_exception_by_middleware(e, request)
+
+        # Complain if the view returned None (a common error).
+        # TODO: New style middlewares will actually be able to handle that, good or not?!
```

Comment:
I'm not sure I see a good use case for it. In any case, this check could be relaxed (or moved to the outer `get_response`, outside the middleware chain) later as a separate change (maybe after the old middleware system is gone, which would reduce the complexity of the change).